Model/Sales/MagentoOrderCollection.php

```diff
@@ -191,6 +209,112 @@ public function setMyParcelTrack()
         return $this;
     }
 
+    /**
+     * Add MyParcel Track from Magento Track
+     *
+     * @return $this
+     * @throws \Exception
+     *
+     */
+    public function setFulfilment()
+    {
+        $apiKey          = $this->getApiKey();
+        $orderCollection = (new OrderCollection())->setApiKey($apiKey);
+        $orderLines      = new Collection();
+
+        foreach ($this->getOrders() as $magentoOrder) {
+            $myparcelDeliveryOptions = $magentoOrder['myparcel_delivery_options'];
+            $deliveryOptions         = json_decode($myparcelDeliveryOptions, true);
+            $deliveryOptionsAdapter  = DeliveryOptionsAdapterFactory::create((array)$deliveryOptions);
+            $this->order             = $magentoOrder;
```

Comment:
waarom deze toewijzing binnen de foreach als je verderop (regel 235, 240, 242) ook gewoon $magentoOrder kunt gebruiken, en er nu na deze foreach een 'onbekende'. order (nl toevallig de laatste) in $this->order zit?